receiver/prometheusexecreceiver/receiver_test.go

```diff
@@ -113,3 +120,663 @@ func assertTwoUniqueValuesScraped(t *testing.T, metricsSlice []pdata.Metrics) {
+							Name:  "DATA_SOURCE_NAME",
+							Value: "user:password@(hostname:{{port}})/dbname",
+						},
+						{
+							Name:  "SECONDARY_PORT",
+							Value: "{{port}}",
+						},
+					},
+				},
+			},
+			newPort: 10111,
+			want: &subprocessmanager.SubprocessConfig{
+				Command: "apache_exporter --port=10111",
+				Env: []subprocessmanager.EnvConfig{
+					{
+						Name:  "DATA_SOURCE_NAME",
+						Value: "user:password@(hostname:10111)/dbname",
+					},
+					{
+						Name:  "SECONDARY_PORT",
+						Value: "10111",
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range fillPortPlaceholdersTests {
+		t.Run(test.name, func(t *testing.T) {
+			got := test.wrapper.fillPortPlaceholders(test.newPort)
+			assert.Equal(t, test.want.Command, got.Command)
+			assert.Equal(t, test.want.Env, got.Env)
+		})
+	}
+}
+
+// Testcases needed for two tests
+var (
+	getDelayAndComputeCrashCountTests = []struct {
+		name               string
+		elapsed            time.Duration
+		healthyProcessTime time.Duration
+		crashCount         int
+		healthyCrashCount  int
+		wantDelay          time.Duration
+		wantCrashCount     int
+	}{
+		{
+			name:               "healthy process 1",
+			elapsed:            15 * time.Minute,
+			healthyProcessTime: 30 * time.Minute,
+			crashCount:         2,
+			healthyCrashCount:  3,
+			wantDelay:          1 * time.Second,
+			wantCrashCount:     3,
+		},
+		{
+			name:               "healthy process 2",
+			elapsed:            15 * time.Hour,
+			healthyProcessTime: 20 * time.Minute,
+			crashCount:         6,
+			healthyCrashCount:  2,
+			wantDelay:          1 * time.Second,
+			wantCrashCount:     1,
+		},
+		{
+			name:               "unhealthy process 1",
+			elapsed:            15 * time.Second,
+			healthyProcessTime: 45 * time.Minute,
+			crashCount:         4,
+			healthyCrashCount:  3,
+			wantCrashCount:     5,
+		},
+		{
+			name:               "unhealthy process 2",
+			elapsed:            15 * time.Second,
+			healthyProcessTime: 75 * time.Second,
+			crashCount:         5,
+			healthyCrashCount:  3,
+			wantCrashCount:     6,
+		},
+		{
+			name:               "unhealthy process 3",
+			elapsed:            15 * time.Second,
+			healthyProcessTime: 30 * time.Minute,
+			crashCount:         6,
+			healthyCrashCount:  3,
+			wantCrashCount:     7,
+		},
+		{
+			name:               "unhealthy process 4",
+			elapsed:            15 * time.Second,
+			healthyProcessTime: 10 * time.Minute,
+			crashCount:         7,
+			healthyCrashCount:  3,
+			wantCrashCount:     8,
+		},
+	}
+	previousResult time.Duration
+)
+
+func TestGetDelay(t *testing.T) {
+	for _, test := range getDelayAndComputeCrashCountTests {
+		t.Run(test.name, func(t *testing.T) {
+			got := getDelay(test.elapsed, test.healthyProcessTime, test.crashCount, test.healthyCrashCount)
+
+			if test.name == "healthy process" {
+				if assert.NotEqual(t, test.wantDelay, got) {
+					assert.Fail(t, "getDelay() got = %v, want %v", got, test.wantDelay)
```

Comment:
I'm not sure what the purpose of this is. This is the only place you use `wantDelay` but you check for the healthy test specifically not having a delay of 1s. Why is that?

Also note you could write this as:
```go
assert.NotEqualf(t, test.wantDelay, got, "getDelay() got = %v, want %v", got, test.wantDelay)
```

Reply:
It's actually that I want it to be equal to 1s, and if it is **not**, then I fail the test and I return, since the rest of the test should not be run. I made this confusing for nothing tbh, I can change it to assert.Equalf(). Also, it's the only test with a deterministic delay, so it's the only one with a `wantDelay`.

Reply:
Makes sense - and looks better now imo 😄